sdk/identity/azure-identity/migration_guide.md

```diff
@@ -0,0 +1,75 @@
+# Guide for migrating to azure-identity from azure-common
+
+## JSON- and file-based authentication
+
+To encourage best security practices, `azure-identity` does not support JSON- and file-based authentication in the same
+way as `azure-common`. `azure-common` provided factory methods like [`get_client_from_json_dict`][client_from_json] and
+[`get_client_from_auth_file`][client_from_auth_file] to provide credentials as plaintext JSON. Azure credentials, like
+passwords, should not be stored in plaintext whenever possible. Instead, `azure-identity` encourages authenticating with
+environment variables (or by other supported means, like [managed identity][managed_identity]).
+
+### Use environment variables
```

Comment:
The section title seems disconnected to the content, and that confused me a bit. I would segment it in three secions

- What you were doing
- What you should do
- What you could do if you need those files